packages/go_router/lib/src/configuration.dart

```diff
@@ -297,8 +297,8 @@ class RouteConfiguration {
   List<RouteMatch>? _getLocRouteMatches(
       Uri uri, Map<String, String> pathParameters) {
     final List<RouteMatch>? result = _getLocRouteRecursively(
-      location: uri.path,
-      remainingLocation: uri.path,
+      location: uri.path.isEmpty ? '/' : uri.path,
```

Comment:
Following are the three call stacks when the breakpoint is set at the above change:
1. On a normal launch
   <details>
   <summary>Call stack</summary>

   ```dart
   RouteConfiguration._getLocRouteMatches (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/configuration.dart:304)
   RouteConfiguration.findMatch (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/configuration.dart:281)
   GoRouteInformationParser.parseRouteInformationWithDependencies (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/parser.dart:87)
   _RouterState._processRouteInformation (.../flutter/stable/packages/flutter/lib/src/widgets/router.dart:744)
   _RouterState.restoreState (.../flutter/stable/packages/flutter/lib/src/widgets/router.dart:622)
   RestorationMixin._doRestore (.../flutter/stable/packages/flutter/lib/src/widgets/restoration.dart:923)
   RestorationMixin.didChangeDependencies (.../flutter/stable/packages/flutter/lib/src/widgets/restoration.dart:909)
   _RouterState.didChangeDependencies (.../flutter/stable/packages/flutter/lib/src/widgets/router.dart:692)
   StatefulElement._firstBuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5448)
   ComponentElement.mount (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5273)
   Element.inflateWidget (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:4182)
   Element.updateChild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:3707)
   ComponentElement.performRebuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5322)
   Element.rebuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5016)
   ComponentElement._firstBuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5279)
   ComponentElement.mount (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5273)
   Element.inflateWidget (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:4182)
   Element.updateChild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:3707)
   ComponentElement.performRebuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5322)
   Element.rebuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5016)
   ```
   </details>
2. Launching deep-link with Uri containing scheme and authority using `adb shell`, when the app is in backgrounded by clicking the home button (To mock deep-link on cold start)
   <details>
   <summary>Call stack</summary>

   ```dart
   RouteConfiguration._getLocRouteMatches (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/configuration.dart:304)
   RouteConfiguration.findMatch (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/configuration.dart:281)
   GoRouteInformationParser.parseRouteInformationWithDependencies (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/parser.dart:87)
   _RouterState._processRouteInformation (.../flutter/stable/packages/flutter/lib/src/widgets/router.dart:744)
   _RouterState.restoreState (.../flutter/stable/packages/flutter/lib/src/widgets/router.dart:622)
   RestorationMixin._doRestore (.../flutter/stable/packages/flutter/lib/src/widgets/restoration.dart:923)
   RestorationMixin.didChangeDependencies (.../flutter/stable/packages/flutter/lib/src/widgets/restoration.dart:909)
   _RouterState.didChangeDependencies (.../flutter/stable/packages/flutter/lib/src/widgets/router.dart:692)
   StatefulElement._firstBuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5448)
   ComponentElement.mount (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5273)
   Element.inflateWidget (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:4182)
   Element.updateChild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:3707)
   ComponentElement.performRebuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5322)
   Element.rebuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5016)
   ComponentElement._firstBuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5279)
   ComponentElement.mount (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5273)
   Element.inflateWidget (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:4182)
   Element.updateChild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:3707)
   ComponentElement.performRebuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5322)
   Element.rebuild (.../flutter/stable/packages/flutter/lib/src/widgets/framework.dart:5016)
   ```
   </details>
3. Launching deep-link with Uri containing scheme and authority using `adb shell`, when the app is in foreground
   <details>
   <summary>Call stack</summary>

   ```dart
   RouteConfiguration._getLocRouteMatches (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/configuration.dart:304)
   RouteConfiguration.findMatch (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/configuration.dart:281)
   GoRouteInformationParser.parseRouteInformationWithDependencies (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/parser.dart:87)
   _RouterState._processRouteInformation (.../flutter/stable/packages/flutter/lib/src/widgets/router.dart:744)
   _RouterState._handleRouteInformationProviderNotification (.../flutter/stable/packages/flutter/lib/src/widgets/router.dart:762)
   ChangeNotifier.notifyListeners (.../flutter/stable/packages/flutter/lib/src/foundation/change_notifier.dart:403)
   GoRouteInformationProvider.notifyListeners (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/information_provider.dart:147)
   GoRouteInformationProvider._platformReportsNewRouteInformation (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/information_provider.dart:246)
   GoRouteInformationProvider.didPushRouteInformation (.../.pub-cache/hosted/pub.dev/go_router-11.1.3/lib/src/information_provider.dart:277)
   WidgetsBinding._handlePushRouteInformation (.../flutter/stable/packages/flutter/lib/src/widgets/binding.dart:704)
   <asynchronous gap> (Unknown Source:0)
   MethodChannel._handleAsMethodCall (.../flutter/stable/packages/flutter/lib/src/services/platform_channel.dart:547)
   <asynchronous gap> (Unknown Source:0)
   _DefaultBinaryMessenger.setMessageHandler.<anonymous closure> (.../flutter/stable/packages/flutter/lib/src/services/binding.dart:567)
   <asynchronous gap> (Unknown Source:0)
   ```
   </details>

You can see, `GoRouteInformationProvider` gets involved only in the 3rd case.

Reply:
for 1 and 2, it would hit this line
https://github.com/flutter/flutter/blob/38f318b46547a6c96715b1d8454c36606acddd42/packages/flutter/lib/src/widgets/router.dart#L619

We should make sure the initialroute is normalized in here
https://github.com/flutter/packages/blob/30b10943f374816454a4d2bf0360cec31b40bca3/packages/go_router/lib/src/router.dart#L523

Reply:
Incorporated suggested changes in the https://github.com/flutter/packages/pull/5113/commits/6cc2fc4f2b7a4ed0801739b4b6fbde77757c96df commit.

Reply:
Before resolving this conversation, please note that the difference between https://github.com/flutter/packages/pull/5113/commits/49f39ad5511e9edcb805c9b18832c0fe32074062 and https://github.com/flutter/packages/pull/5113/commits/6cc2fc4f2b7a4ed0801739b4b6fbde77757c96df is that former tries to fix the issue as late as possible and later as early as possible.
This might create issues in the future when the package handles other schemes like `file:///`.